solr/api/src/java/org/apache/solr/client/api/endpoint/CollectionApis.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.client.api.endpoint;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.QueryParam;
+import org.apache.solr.client.api.model.CreateCollectionRequestBody;
+import org.apache.solr.client.api.model.ListCollectionsResponse;
+import org.apache.solr.client.api.model.ModifyCollectionRequestBody;
+import org.apache.solr.client.api.model.SubResponseAccumulatingJerseyResponse;
+
+/** V2 API definitions for manipulating Solr collections */
+public interface CollectionApis {
+
+  @Path("/collections")
+  interface List {
+    @GET
+    @Operation(
+        summary = "List all collections in this Solr cluster",
+        tags = {"collections"})
+    ListCollectionsResponse listCollections();
+  }
+
+  /** V2 API definition for creating a SolrCloud collection */
+  @Path("/collections")
+  interface Create {
+    @POST
+    @Operation(
+        summary = "Creates a new SolrCloud collection.",
+        tags = {"collections"})
+    SubResponseAccumulatingJerseyResponse createCollection(CreateCollectionRequestBody requestBody)
+        throws Exception;
+  }
+
+  @Path("/collections/{collectionName}")
+  interface Modify {
+    @PUT
```

Comment:
BTW I'm suspicious of your proposed use of HTTP **PUT** as the payload isn't an entire representation of the target resource.  It modifies an existing one that may have characteristics specified on creation that are not here.  PUT doesn't have much applicability.

[1] https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT "**replaces** a representation of the target resource with the request content"
[1] https://restfulapi.net/rest-put-vs-post/ "PUT **replaces** the resource in its entirety"

Reply:
I've proposed using the `PATCH` verb for cases like this - seemingly it's made for cases like this.  But I've gotten feedback in the past that it's unintuitive due to its obscurity.  Wdyt?

Reply:
I've used PATCH in other contexts.    I *think* that if we embrace PATCH only here, then yeah, it will be kind of obsurcure.   But if we use it on multiple APIs, then I think it's perfectly fine.

Reply:
[PATCH](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH) makes total sense to me.  I don't care about relative obscurity.

Reply:
> I think that if we embrace PATCH only here, then yeah, it will be kind of obsurcure. But if we use it on multiple APIs, then I think it's perfectly fine.

That seems like a reasonable line to walk here.  Anyone have other APIs in mind that use similar partial-update semantics?  If so, I'll switch this to 'PATCH'.  Otherwise I'll stick with 'PUT' for now.